### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by proloyd**
FIX Avoid divide by zero in vector.pyx

### CODE REVIEW

File: eelbrain/_stats/vector.pyx [hunk 1]
```diff
@@ -137,11 +137,11 @@ def t2_stat(cnp.ndarray[FLOAT64, ndim=3] y,
                 sigma[u][v] -= mean[u] * mean[v] / n_cases
         # check non-zero variance
         for v in range(n_dims):
-            if sigma[v][0] != 0:
+            if sigma[v][v] != 0.0:
                 break
-        else:
-            out[i] = 0
-            continue
+            else:
```
Comment (christianbrodbeck) [inline eelbrain/_stats/vector.pyx]: The indent was correct, it's `for` loop syntax, a statement to execute only it does not break

**Comment by proloyd** [comment]
My bad! Oops. I was not aware of that.

File: eelbrain/_stats/vector.pyx [hunk 2]
```diff
@@ -137,7 +137,7 @@ def t2_stat(cnp.ndarray[FLOAT64, ndim=3] y,
                 sigma[u][v] -= mean[u] * mean[v] / n_cases
         # check non-zero variance
         for v in range(n_dims):
-            if sigma[v][0] != 0:
+            if sigma[v][v] != 0:
```
Comment (christianbrodbeck) [inline eelbrain/_stats/vector.pyx:140]: Now you're checking the diagonal...

File: eelbrain/_stats/vector.pyx [hunk 3]
```diff
@@ -151,11 +151,12 @@ def t2_stat(cnp.ndarray[FLOAT64, ndim=3] y,
...
-            if eig[v] > TOL:
-                norm += temp ** 2 / eig[v]
-            else:
-                # norm += 0
-                norm += temp ** 2 / TOL
+            if temp != 0.0:     # Avoid divide by zero
+                if eig[v] > TOL:
+                    norm += temp ** 2 / eig[v]
+                else:
+                    # norm += 0
```
Comment (christianbrodbeck) [inline eelbrain/_stats/vector.pyx]: This comment can be removed right?

File: eelbrain/_stats/vector.pyx [hunk 4]
```diff
@@ -199,7 +200,7 @@ def t2_stat_rotated(cnp.ndarray[FLOAT64, ndim=3] y,
                 sigma[v][u] -= mean[u] * mean[v] / n_cases
         # check non-zero variance
         for v in range(n_dims):
-            if sigma[v][0] != 0:
+            if sigma[v][v] != 0:
```
Comment (christianbrodbeck) [inline eelbrain/_stats/vector.pyx:202]: same as above

File: eelbrain/_stats/vector.pyx (hunk 2 above)
Comment (proloyd) [reply eelbrain/_stats/vector.pyx:140]: Yes! Preciously. If all three diagonals are zero, the matrix cannot be semi-definite (i.e. a must have property for co-variance matrices). By the way, these diagonals are variances.

Comment (christianbrodbeck) [reply eelbrain/_stats/vector.pyx:140]: Haha sorry my bad :)

**Comment by christianbrodbeck** [comment]
Thanks!

**Comment by proloyd** [comment]
No worries!

**Pull request merged**